src/app/components/SideBar.tsx

```diff
@@ -62,6 +62,43 @@ const SideBar: React.FC<SideBarProps> = ({
     routeType,
 }) => {
     const selectedProfAndCourse = selectedProfessor && selectedCourse;
+    const [yearChangedByUser, setYearChangedByUser] = React.useState(false);
+    
```

Comment:
For now followed some timeline to define semesters (you can change them how you like)
Spring Semester: January 1st to May 15th.
Summer Semester: May 16th to August 15th.
Fall Semester: August 16th to December 20th.